trin-core/src/utp/stream.rs

```diff
@@ -117,14 +103,9 @@ pub struct UtpListener {
     /// Base discv5 layer
     discovery: Arc<Discovery>,
     /// Store all active connections
-    utp_connections: HashMap<ConnectionKey, UtpSocket>,
+    utp_connections: HashMap<NodeId, UtpSocket>,
```

Comment:
It seems that this change would prevent more than one simultaneous uTP connection between the local node and some remote peer. This would be problematic if the local node and remote peer participate in multiple portal subnetworks.

I would need to investigate further to see whether this is possible, but another concern is the following scenario:

- The local node initiates a uTP connection with some remote peer on subnetwork `X`
- The local node and remote peer begin to exchange data
- The remote peer initiates a uTP connection with the local node on subnetwork `Y`
- The second connection disrupts the first connection by taking over its entry in the connection map

Either the uTP connection on subnetwork `Y` fails or it hijacks the existing uTP connection on subnetwork `X`.

Perhaps I'm missing something here though.

Reply:
Good call, I brought back the `ConnectionKey` struct which is a combination of `NodeId` and uTP socket receive connection id. As we generate those connection ids randomly, it is a very low probability to have two active uTP connections with the same connection id and `NodeId`.